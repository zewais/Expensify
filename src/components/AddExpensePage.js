import React from "react";
import ExpenseForm from "./ExpenseForm"
import {connect} from "react-redux"
import {addExpense} from "../actions/expenses"



const AddExpensePage = (props) => (
    <div>
        <h1> Add Expense </h1>
        <ExpenseForm 
            onSubmit={(expense) => {
                console.log(expense)
                props.dispatch(addExpense(expense))
            }}
        />
    </div>
    
);

export default connect()(AddExpensePage);