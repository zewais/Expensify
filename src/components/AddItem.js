import React from "react";

class AddItem extends React.Component {
    state = {
        error: undefined
    }


    handleAddItem = (e) => {

        e.preventDefault();

        const item = e.target.itemEntered.value.trim();
        const error = this.props.handleAddItem(item)

        this.setState(() => ({error}))

        if(!error){
            e.target.itemEntered.value = "";
        }

    }

    render() {
        return (
            <div>
                {this.state.error && <p className="add-item-error">{this.state.error}</p>}
                <form onSubmit = {this.handleAddItem} className="add-item">
                    <input type="text" name="itemEntered" className="add-item__input"/>
                    <button className="button">Add item</button>
                </form>

            </div>
        )
    }
}


export default AddItem;