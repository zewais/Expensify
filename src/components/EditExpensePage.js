import React from "react";
import ReactDOM from "react-dom";
import {BrowserRouter, Route, Switch, Link, NavLink} from "react-router-dom";
import {connect} from "react-redux";
import ExpenseForm from "./ExpenseForm"
import {startRemoveExpense, startEditExpense} from "../actions/expenses"


export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense)
        this.props.history.push("/")
    }

    onClick = () => {
        this.props.startRemoveExpense({id: this.props.expense.id})
        this.props.history.push("/")
    }

    render() {
        return(
        <div>
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title"> Edit Expense </h1>
            </div>
        </div>
        <div className="content-container">
        <ExpenseForm 
        expense={this.props.expense}
        onSubmit = {this.onSubmit}
        />
        <button className="button-layout__remove" onClick = {this.onClick}>Remove Expense</button>
        </div>
        </div>
            
            
        
        )
    }
}


/*const EditExpensePage = (props) => {
    console.log(props);
    return (
        <div>
            <ExpenseForm 
            expense={props.expense}
                onSubmit = {(expense) => {
                    props.dispatch(editExpense(props.expense.id, expense))
                    props.history.push("/")

                }}
            />
            <button onClick = {(e) => {
                props.dispatch(removeExpense({id: props.expense.id}))
                props.history.push("/")
            }}>Remove</button>
            
        </div>
    );
}*/


const mapStateToProps = (state,  props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)

    }
        
}

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);

