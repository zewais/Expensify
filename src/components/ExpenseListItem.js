import React from "react";
import { removeExpense } from "../actions/expenses";
import {connect} from "react-redux";
import {Link} from "react-router-dom"
import moment from "moment"
import numeral from "numeral"

const ExpenseListItem = ({dispatch, id, description, amount, createdAt}) => (
    
   
       <Link className="list-item" to={`/edit/${id}`}>
       <div>
       <h3 className="list-item__title">{description}</h3>
       <span className="list-item__sub-title">{moment(createdAt).format("MMMM Do, YYYY")}</span>
       </div>
       <div>
       <h3 className="list-item__data">{numeral(amount/100).format("$0,0.00")}</h3>
       </div>
       </Link>
        
        
   
)

export default ExpenseListItem;