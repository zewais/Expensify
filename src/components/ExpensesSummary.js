import React from "react"
import {connect} from "react-redux"
import numeral from "numeral"
import selectExpenses from "../selectors/expenses"
import selectExpensesTotal from "../selectors/expenses-total"
import {Link} from "react-router-dom"

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    
    return(
        <div className="page-header">
        <div className="content-container">
            <h1 className="page-header__title">Viewing <span>{expensesCount}</span> {expensesCount === 1? "expense": "expsenses"} with total of <span>{numeral(expensesTotal).format("$0,0.00")}</span></h1>
            <div className="page-header__actions">
                <Link className="button-layout" to="/create">Add Expense</Link>
            </div>
        </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visableExpenses = selectExpenses(state.expenses, state.filters)

    return {
        expensesCount: visableExpenses.length,
        expensesTotal: selectExpensesTotal(visableExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)