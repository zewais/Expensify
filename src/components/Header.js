import React from "react";
import ReactDOM from "react-dom";
import {BrowserRouter, Route, Switch, Link, NavLink} from "react-router-dom";
import { connect } from "react-redux"
import {startLogout} from "../actions/auth"


export const Header = ({startLogout}) => (
    <header className="header">
    <div className="content-container">
    <div className="header__content">
        <Link className="header__title" to="/dashboard" exact={true}>
            <h1>Expensify</h1>
        </Link>
        <button className="button-layout button-layout__link" onClick={startLogout}>Logout</button>
    </div>
    </div>
        
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);