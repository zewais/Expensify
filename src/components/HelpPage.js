import React from "react";
import ReactDOM from "react-dom";
import {BrowserRouter, Route, Switch, Link, NavLink} from "react-router-dom";


const HelpPage = () => (
    <div>
        This is the help page
    </div>
)

export default HelpPage;