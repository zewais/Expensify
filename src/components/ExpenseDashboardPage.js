import React from "react";
import ReactDOM from "react-dom";
import ExpenseList from "./ExpenseList"
import ExpenseListFilters from "./ExpenseListFilters"
import ExpensesSummary from "./ExpensesSummary"

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;
