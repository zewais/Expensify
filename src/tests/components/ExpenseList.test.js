import {ExpenseList} from "../../components/ExpenseList"
import React from "react"
import {shallow} from "enzyme"
import expenses from "../fixtures/expenses"


test("should render ExpenseList with expenses", () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot();
})

test("should render ExpenseList with no expense message", () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>)
    expect(wrapper).toMatchSnapshot();
})