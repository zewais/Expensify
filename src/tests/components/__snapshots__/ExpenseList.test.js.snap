// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseListItem
      amount={150}
      createdAt={0}
      description="gum"
      id="1"
      key="1"
      note=""
    />
    <ExpenseListItem
      amount={109500}
      createdAt={-345600000}
      description="rent"
      id="2"
      key="2"
      note=""
    />
    <ExpenseListItem
      amount={4500}
      createdAt={345600000}
      description="Credit Card"
      id="3"
      key="3"
      note=""
    />
  </div>
</div>
`;

exports[`should render ExpenseList with no expense message 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item__message"
    >
      <span>
        No Expenses
      </span>
    </div>
  </div>
</div>
`;
